### DISCUSSION TITLE
resampleImage

### DISCUSSION THREAD

**Discussion by fabrice4821** [ImageMagick/ImageMagick#2483] [Help]
Hi,

Normally, I use the command : `exec('convert -density 144 .......');` that works fine.
But on a server, I can't use the command exec.

I try : 
	$image = new Imagick();
	$image->readimage($pdf_file);
	$image->setImageResolution(144,144);
	$image->resampleImage(144,144,imagick::FILTER_UNDEFINED,0);
	$image->writeImage($pdf_file.'.jpg'); 

But, this command doesn't work !
My file is not resample.

What is my error ?

**Comment by dlemstra** [comment]
My guess would be that you should not call `$image->setImageResolution(144,144);` before you do the call `$image->resampleImage`. The first one adjusts the density and the second one will not resample because the density matches the current density. But that is just a guess the @ImageMagick team does not write or maintain the `IMagick` library.

**Reply by fmw42** [reply]
Set the resolution before you read the PDF.

**Comment by fabrice4821** [comment]
Thanks, but it doesn't work.
